plan/relations.go

```diff
@@ -156,6 +176,24 @@ func (n *NamedTableReadRel) ToProto() *proto.Rel {
 	}
 }
 
+func (n *NamedTableReadRel) Copy(newInputs ...Rel) (Rel, error) {
+	return n, nil
+}
+
+func (n *NamedTableReadRel) CopyWithExpressionRewrite(rewriteFunc func(expr.Expression) (expr.Expression, error), newInputs ...Rel) (Rel, error) {
+	newExprs, err := n.baseReadRel.copyExpressions(rewriteFunc)
+	if err != nil {
+		return nil, err
+	}
+	if newExprs == nil {
+		return nil, nil
```

Comment:
per comment above, let's avoid returning nil in the case that expressions are unchanged and return original.